spec.html

```diff
@@ -2,20 +2,490 @@
+      </ul>
+      <p>The default implementation of HostMakeJobCallback performs the following steps when called:</p>
+      <emu-alg>
+        1. <del>Return the JobCallback Record { [[Callback]]: _callback_, [[HostDefined]]: ~empty~ }.</del>
+        1. <ins>Let _snapshot_ be AsyncContextSnapshot().</ins>
+        1. <ins>Return the JobCallback Record { [[Callback]]: _callback_, [[AsyncContextSnapshot]]: _snapshot_, [[HostDefined]]: ~empty~ }.</ins>
+      </emu-alg>
+      <p>ECMAScript hosts that are not web browsers must use the default implementation of HostMakeJobCallback.</p>
+      <emu-note>
+        <p>HostMakeJobCallback snapshots the current AsyncContext global state. HostCallJobCallback restores the snapshot.</p>
+      </emu-note>
+    </emu-clause>
+
+    <emu-clause id="sec-hostcalljobcallback" type="host-defined abstract operation">
+      <h1>
+        HostCallJobCallback (
+          _jobCallback_: a JobCallback Record,
+          _V_: an ECMAScript language value,
+          _argumentsList_: a List of ECMAScript language values,
+        ): either a normal completion containing an ECMAScript language value or a throw completion
+      </h1>
+      <dl class="header">
+      </dl>
+      <p>An implementation of HostCallJobCallback must conform to the following requirements:</p>
+      <ul>
+        <li><ins>It must perform AsyncContextSwap before and after the call,</ins></li>
+        <li>and perform and return the result of Call(_jobCallback_.[[Callback]], _V_, _argumentsList_).</li>
+      </ul>
+      <p>The default implementation of HostCallJobCallback performs the following steps when called:</p>
+      <emu-alg>
+        1. Assert: IsCallable(_jobCallback_.[[Callback]]) is *true*.
+        1. <del>Return ? Call(_jobCallback_.[[Callback]], _V_, _argumentsList_).</del>
+        1. <ins>Let _F_ be _jobCallback_.[[Callback]].</ins>
+        1. <ins>Let _snapshot_ be _jobCallback_.[[AsyncContextSnapshot]].</ins>
+        1. <ins>Let _asynccontextMapping_ be AsyncContextSwap(_snapshot_).</ins>
+        1. <ins>Let _result_ be Completion(Call(_F_, _V_, _argumentsList_)).</ins>
+        1. <ins>AsyncContextSwap(_asynccontextMapping_).</ins>
+        1. <ins>Return _result_.</ins>
+      </emu-alg>
+      <p>ECMAScript hosts that are not web browsers must use the default implementation of HostCallJobCallback.</p>
+      <emu-note>
+        <p>HostCallJobCallback restores the snapshot saved in HostMakeJobCallback when calling the _jobCallback_.[[Callback]].</p>
+      </emu-note>
+    </emu-clause>
+  </emu-clause>
+</emu-clause>
+
+<emu-clause id="sec-ordinary-and-exotic-objects-behaviours">
+  <h1>Ordinary and Exotic Objects Behaviours</h1>
+  <emu-clause id="sec-built-in-exotic-object-internal-methods-and-slots">
+    <h1>Built-in Exotic Object Internal Methods and Slots</h1>
+
+    <emu-clause id="sec-asynccontext-wrapped-function-exotic-objects">
+      <h1>AsyncContext Wrapped Function Exotic Objects</h1>
```

Comment:
Note to self: this is based on https://tc39.es/ecma262/#sec-bound-function-exotic-objects.